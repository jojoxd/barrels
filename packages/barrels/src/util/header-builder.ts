import {BarrelContext} from "../model/barrel-context.js";

export class HeaderBuilder
{
    public static buildHeader(context: BarrelContext): string
    {
        const headerLines = context.header.split(/\r?\n/g);

        // @TODO: Add line endings to configuration, see also ast-transformer
        return `/**\n * ${headerLines.join('\n * ')}\n */\n`;
    }
}
