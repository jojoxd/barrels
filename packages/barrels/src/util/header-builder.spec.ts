import { HeaderBuilder } from "./header-builder.js";
import {createBarrelsConfiguration} from "../../tests/support/create-barrels-configuration.js";

const testHeaders = [
    'Test',
    '@generated by @jojoxd/barrels',
    undefined,
    'Multiline\nTest',
    'Multiline Windows\r\nTest',
];

it.each(testHeaders)('Should generate a header', async (testHeader) => {
    const { context, } = await createBarrelsConfiguration({
        barrels: [{ directory: 'empty' }],
        header: testHeader,
    });

    const header = HeaderBuilder.buildHeader(context);

    expect(header).toMatchSnapshot();
});