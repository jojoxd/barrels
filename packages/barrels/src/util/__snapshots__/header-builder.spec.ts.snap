// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should generate a header 1`] = `
"/**
 * Test
 */
"
`;

exports[`Should generate a header 2`] = `
"/**
 * @generated by @jojoxd/barrels
 */
"
`;

exports[`Should generate a header 3`] = `
"/**
 * @generated by @jojoxd/barrels
 */
"
`;

exports[`Should generate a header 4`] = `
"/**
 * Multiline
 * Test
 */
"
`;

exports[`Should generate a header 5`] = `
"/**
 * Multiline Windows
 * Test
 */
"
`;
